app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,66 @@
+class TasksController < ApplicationController
+  before_action :find_user, only: %i[new create index show destroy update find_task]
+  before_action :find_task, only: %i[show destroy update]
+  before_action :require_login
+  load_and_authorize_resource
+
+  def index
+    @tasks = Task.all
```

Comment:
Can we try to apply this instead of fetching all tasks? 
https://github.com/CanCanCommunity/cancancan#fetching-records